### DISCUSSION TITLE
.Net 7 WPF Single File Application crashes

### DISCUSSION THREAD

**Discussion by forReason** [cefsharp/CefSharp#4376] [General]
- **What version of the product are you using?** 
`109.1.110`
- Nuget Package: 
`CefSharp.Wpf.NETCore`
- **What architecture x86 or x64?** 
tried both, target is `x64`
- **What version of .Net?** 
`.Net 7`
- **On what operating system?** 
`Win 10 22h2`
- **Are you using `WinForms`, `WPF` or `OffScreen`?** 
`WPF`
    
- **What steps will reproduce the problem?**
1. Create a .Net 7 WPF Project
2. Add a CefSharp Control
3. Publish single file
4. run .exe

- **What is the expected output? What do you see instead?**
-> application does not crash
-> Application crashes, debug information unavailable due to missing symbols exception returning some memory locations in excep.cpp

- **Please provide any additional information below.**
The problem with single File Publish is known from earlier versions where I found some issues and documentation.
The solution is supposedly to selfhost the BrowserSubprocess from within the applications main entrance point.

In .Net 7, the main entrance point is defined in xaml so I could not find how I would host my own BrowserSubprocess.
The most helpful answer so far was in [Publish self-contained .NET 5 application](https://github.com/cefsharp/CefSharp/issues/3407#top)
#3407.

I tried to add the code from https://github.com/cefsharp/CefSharp.MinimalExample/blob/master/CefSharp.MinimalExample.Wpf/App.xaml.cs to my app.xaml.cs without success.
I attached the exception I get:
[errors.log](https://github.com/cefsharp/CefSharp/files/10501125/errors.log)

Trying the minimal example and changing its target framework wont work either. 
- publish wont produce a single file but compile into many files
- The application crashes on start
- no log file is created

**Comment by amaitland** [comment]
> * The solution is supposedly to selfhost the BrowserSubprocess from within the applications main entrance point.

`SelfHosting` the `BrowserSubprocess` is **required** for publish single file, there's no `BrowserSubProcess` included in the build.

> In .Net 7, the main entrance point is defined in xaml so I could not find how I would host my own BrowserSubprocess.

That's the same for all `WPF` applications by default, you simply need to add your own. 

https://stackoverflow.com/questions/6156550/replacing-the-wpf-entry-point has generic guidance.

> I tried to add the code from https://github.com/cefsharp/CefSharp.MinimalExample/blob/master/CefSharp.MinimalExample.Wpf/App.xaml.cs to my app.xaml.cs without success.

https://github.com/cefsharp/CefSharp.MinimalExample/blob/master/CefSharp.MinimalExample.Wpf/Program.PublishSingleFile.cs provides an example.


> Trying the minimal example and changing its target framework wont work either.

The tested/recommended publish command is listed at https://github.com/cefsharp/CefSharp/issues/3407#issuecomment-787008626

The minimal example should work for both `.net5.0` and `.net6.0`, minor changes to the csproj would be required to add `.net7.0`
